pkg/pillar/cmd/upgradeconverter/upgradeconverter.go

```diff
@@ -0,0 +1,68 @@
+package upgradeconverter
+
+import (
+	"flag"
+	"github.com/lf-edge/eve/pkg/pillar/pubsub"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"log"
+)
+
+type upgradeconverterContext struct {
+	subGlobalConfig pubsub.Subscription
+	globalConfig    *types.ConfigItemValueMap
+}
+
+type ConverterAction uint8
+
+const (
+	Convert ConverterAction = iota + 1
+	CleanupOld
+	CleanupNew
+	NoAction
+)
+
+func Run() {
+	action := Convert
+	convert := *flag.Bool("convert", false, "Convert from old to new")
+	cleanupOld := *flag.Bool("cleanupOld", false, "Cleanup old versions")
+	cleanupNew := *flag.Bool("cleanupNew", false, "Cleanup new versions")
+	flag.Parse()
+	if cleanupOld {
+		action = CleanupOld
+	} else if cleanupNew {
+		action = CleanupNew
+	} else if convert {
+		action = Convert
+	} else {
+		action = NoAction
+	}
+
+	conversionHandlers := []ConversionHandler{
+		{
+			description: "",
+			handlerFunc: convertGlobalSettingsToUseAttrValueMap,
+		},
+	}
+	for _, handler := range conversionHandlers {
+		err := handler.handlerFunc(action)
+		log.Fatalf("An error occured %s", err)
+	}
+
+}
+
+type HandlerFunc func(action ConverterAction) error
+
+type ConversionHandler struct {
+	description string
+	handlerFunc HandlerFunc
+}
+
+func convertGlobalSettingsToUseAttrValueMap(action ConverterAction) error {
+	switch action {
+	case Convert:
+	case CleanupNew:
+	case CleanupOld:
+	case NoAction:
+	}
+	return nil
+}
```

Comment:
Best to put this in a separate file.. Each handler can be in a separate file ( But same package ).. It will be clean then..